switcheroo/server/__main__.py

```diff
@@ -0,0 +1,36 @@
+from argparse import ArgumentParser
+from pathlib import Path
+from switcheroo.server.retrieve_public_keys import get_public_keys_local, get_public_keys_s3
+
+def create_argument_parser() -> ArgumentParser:
+    argument_parser = ArgumentParser(
+        prog="key_retriever",
+        description="Fetches the public SSH keys from S3 or the local machine",
+        epilog="Thanks for using key_retriever! :)"
+    )
+    argument_parser.add_argument("user")
+    argument_parser.add_argument(
+        "-ds",
+        "--datastore",
+        choices=["s3", "local"],
+        default="s3",
+        help="choose where to fetch the public key from, S3 or the local system (default is S3)",
+    )
+    argument_parser.add_argument(
+        "--bucketname",
+        required=False,
+        help="If s3 is selected, the bucket name to look for the key",
+    )
+    return argument_parser
+
+if __name__ == "__main__":
+    parser = create_argument_parser()
```

Comment:
put this logic in its own function (`main`) and make sure we have a proper try/except to catch an errors that might occur and produce a proper error message in the end

Reply:
the client lost connectivity, AWS is offline,. key in question doesn't exist, there are literally millions of things that can go wrong